sorting/merge_sort.c

```diff
@@ -33,6 +33,8 @@ void swap(int *a, int *b)
 void merge(int *a, int l, int r, int n)
 {
     int *b = (int *)malloc(n * sizeof(int)); /* dynamic memory must be freed */
+    if (!b) /* return to main if can't malloc memory */
+        return ;
```

Comment:
Maybe you should just make it print a message then exit, it's not like it can resume normal operation afterwards anyway